src/main/java/com/develhope/spring/vehicles/controller/VehicleController.java

```diff
@@ -13,13 +15,44 @@
 @RestController
 @RequestMapping("/vehicle")
 public class VehicleController {
+
     @Autowired
     private VehicleService vehicleService;
 
     @PostMapping
     public ResponseEntity<Vehicle> createVehicle(@RequestBody Vehicle vehicle) {
-        Vehicle vehicleAdded = vehicleService.createVehicle(vehicle);
-        return ResponseEntity.ok(vehicleAdded);
+        try {
+            Vehicle vehicleAdded = vehicleService.createVehicle(vehicle);
+            return ResponseEntity.ok(vehicleAdded);
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @PostMapping("/shop")
+    public ResponseEntity<String> shopVehicle(@RequestBody Vehicle vehicle, @RequestParam VehicleState isPurchasable) {
+        try {
+            boolean isPurchased = vehicleService.shopVehicle(vehicle, isPurchasable);
+            if (isPurchased) {
+                return ResponseEntity.ok("Vehicle has been successfully purchased!");
+            }
+            return ResponseEntity.badRequest().body("Vehicle purchase failed, please call our agency for more information.");
+        } catch (OrderNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        }
+    }
+
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<String> findPurchasedVehiclesByUserId(@PathVariable Long userId, Order order) {
```

Comment:
Perché prende un ordine in ingresso? Non dovrebbe andare a vedersi tutti gli ordini fatti da quell'utente e restituirci una lista dei veicoli?